sdk/rust/oak_utils/src/lib.rs

```diff
@@ -0,0 +1,116 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use std::collections::HashMap;
+use std::fs;
+use std::io;
+use std::path::Path;
+
+#[cfg(test)]
+mod tests;
+
+// Generates Rust `proto` files in a temporary directory using `protoc_rust`,
+// checks previously generated files and updates them if their contents have changed.
+// This is a workaround for `protoc_rust` that always updates files, thus provoking recompilation
+// of all dependent targets.
```

Comment:
Could you add a link to https://github.com/rust-lang/cargo/issues/6529 here or somewhere else?